### DISCUSSION TITLE
Package composition and `viewMode: 'docs'`

### DISCUSSION THREAD

**Discussion by benedfit** [storybookjs/storybook#16935] [Help] [answered]
Hi, I have a shared UI components package (https://6004028a0df11e0021f58df8-zbpsazzvsf.chromatic.com/), that is exposed for composition via https://github.com/newhighsco/chipset/blob/main/package.json#L25, and is consumed by another repository. Package composition works as expected (https://61531de3bb5649003ab4f605-luacxyskfl.chromatic.com/?path=/docs/components-avatar--source), but with one minor issue. I'm running the consuming Storybook in `viewMode: 'docs'`, however if I click on one of the stories for the shared package it shows each of the individual stories in the navigation.

Before I raised a new issue for this, I wanted to check if there is some additional configuration that I need to add the get the desired effect?

**Comment by ddamato** [comment]
I'm actually trying to do _exactly_ what you've demonstrated but I'm unfortunately getting exactly what you want. The big difference is that I'm trying to build two storybook's from the same repository. 🙃 

Admittedly, I don't want to do this because I'd much rather have some stories be presented only as docs and others presented only as stories. I thought I could use composition to "lock" the presentational stories into a "no docs mode" to be displayed in a docs mode host storybook but everything I've tried makes the stories-only completely missing, either from the sidebar or broken as a link when messing with the URL params. I would have also considered trying to _force_ the URL to use `viewMode=story` but it looks like it doesn't want to show the stories.

It would be nice to have some way of telling `docsMode` to allow some stories to be only stories and not docs or vice versa. Perhaps setting `viewMode: 'story'` makes sense as an enhancement or maybe introducing `docsMode: false` at a story?

**Comment by shilman** [comment, marked as answer]
Late to the party but try this solution in recent versions of Storybook (8.4+)

https://storybook.js.org/docs/writing-stories/tags#docs-only-stories

**Reply by ddamato** [reply]
Actually that works! I'm surprised how powerful these tags are!
